lib/internal/bootstrap_node.js

```diff
@@ -559,9 +563,30 @@
     nativeModule.cache();
     nativeModule.compile();
 
+    if (config.experimentalModules) {
+      NativeModule._nsCache[id] = {
+        default: nativeModule.exports,
+      };
+
+      const descriptors =
+        Object.getOwnPropertyDescriptors(nativeModule.exports);
+      for (const [name, d] of Object.entries(descriptors)) {
+        if (d.set !== undefined || d.get !== undefined || !d.enumerable)
+          continue;
+        if (/^_|_$/.test(name))
```

Comment:
Why add the `_` filter? If it can me accessed via CJS on a `module.export` it should be accessible as a named export.

Reply:
all the underscore exports are undocumented methods, usually exported for tests (which is another issue i want to address in a separate pr) and i see 0 reason for them to be added to the module namespace.

Reply:
@jdalton Always feel free to file an issue if you want the functionality of an underscored property exposed!

Reply:
Named exports is named exports. Adding artificial filters, just as with `Module.builtinModules` *which I successfully [pushed back again](https://github.com/nodejs/node/pull/16386#issuecomment-347903434)*, isn't great. If something isn't meant to be exposed then it shouldn't be exported and it should go through the proper process of deprecation and removal.